iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -582,6 +582,11 @@ public void validateSinkDesc(FileSinkDesc sinkDesc) throws SemanticException {
     HiveStorageHandler.super.validateSinkDesc(sinkDesc);
     if (sinkDesc.getInsertOverwrite()) {
       Table table = IcebergTableUtil.getTable(conf, sinkDesc.getTableInfo().getProperties());
+      if (table.currentSnapshot() != null &&
+          "0" .equalsIgnoreCase(table.currentSnapshot().summary().get(SnapshotSummary.TOTAL_RECORDS_PROP))) {
```

Comment:
can we cast to long and compare numbers?
````
Long.parseLong(table.currentSnapshot().summary().get(SnapshotSummary.TOTAL_RECORDS_PROP)) == 0
````

Reply:
should we move the check to line 594?

Reply:
Changed to compare to Long.
Regarding moving the check below, I don't think we need to throw in case of bucketed tables as well, if we establish that the table is empty. That check was added as part of [HIVE-25849](https://issues.apache.org/jira/browse/HIVE-25849) in order to prevent overwriting wrong data or duplications but in case we establish the table is empty, this no longer holds true

https://github.com/apache/hive/pull/2856#issue-1074445457